core-test/src/test/scala/org/bitcoins/core/psbt/PSBTTest.scala

```diff
@@ -310,6 +310,22 @@ class PSBTTest extends BitcoinSAsyncTest {
     }
   }
 
+  it must "add Redeem Scripts and Witness Scripts to outputs" in {
+    forAllAsync(PSBTGenerators.fullNonFinalizedPSBT,
+                ScriptGenerators.randomNonP2SHScriptPubKey,
+                WitnessGenerators.scriptWitness) {
+      case (psbtF, spk, wit) =>
+        psbtF.map { psbt =>
+          val randOutput = scala.util.Random.nextInt(psbt.outputMaps.length)
+
+          psbt.addRedeemOrWitnessScriptToOutput(spk._1, randOutput)
+          psbt.addScriptWitnessToOutput(wit, randOutput)
```

Comment:
I'm confused as to why this should work, won't this cause things to be inconsistent as they are random?